File: code/Programmers/Level2/카펫/카펫_김수진.java
class Solution {
    public int[] solution(int brown, int red) {
        int[] answer = {0,0};
        
        for(int i=1;i<=red;i++){
            int redWidth = red/i;
            int redHeight = i;
            
            int brownNum = (redWidth+2) * (redHeight+2);
            
            if((brownNum-red) <= brown){
                answer[0] = redWidth+2;
                answer[1] = redHeight+2;
                return answer;
            }
        }
        return answer;
    }
}

File: code/Programmers/Level2/카펫/카펫_김수진.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer; //가로, 세로 
    int total = brown + yellow; 
    int minH = 3;
    int minW = 0;
    while(1){
        minW = (brown - minH*2)/2 +2;
        if((minW-2)*(minH-2) == yellow)
            break;
        else minH++;
    }
    answer.push_back(minW);
    answer.push_back(minH);
    return answer;
}
